From 30ebcc82ad5d148e26336ad4b90520080cda63b8 Mon Sep 17 00:00:00 2001
From: Bogdan-Gabriel Roman <bogdan-gabriel.roman@nxp.com>
Date: Tue, 22 Feb 2022 22:13:17 +0200
Subject: [PATCH 19/51] s32: remove SRAM code

Remove SRAM code. Delete files which are rendered empty due to this
removal.

Issue: ALB-8342
Upstream-Status: Pending 

Signed-off-by: Bogdan-Gabriel Roman <bogdan-gabriel.roman@nxp.com>
Signed-off-by: Zhantao Tang <zhantao.tang@windriver.com>
---
 arch/arm/cpu/armv8/s32/cpu.c                |  1 -
 arch/arm/cpu/armv8/s32/s32-gen1/Makefile    |  1 -
 arch/arm/cpu/armv8/s32/s32-gen1/s32g274a.c  | 60 ---------------------
 arch/arm/cpu/armv8/s32/s32-gen1/soc.c       |  1 -
 arch/arm/include/asm/arch-s32/soc.h         | 20 -------
 board/freescale/s32-gen1/common.c           |  1 -
 board/freescale/s32-gen1/eth.c              |  1 -
 board/freescale/s32-gen1/s32g274abluebox3.c |  1 -
 board/freescale/s32-gen1/s32gxxxaevb.c      |  1 -
 board/freescale/s32-gen1/s32gxxxardb.c      |  1 -
 board/freescale/s32-gen1/s32r45evb.c        |  1 -
 drivers/net/dwc_eth_qos_s32cc.c             |  1 -
 drivers/net/pfeng/pfeng.h                   |  4 ++
 drivers/net/pfeng/pfeng_cmd.c               |  1 -
 14 files changed, 4 insertions(+), 91 deletions(-)
 delete mode 100644 arch/arm/cpu/armv8/s32/s32-gen1/s32g274a.c
 delete mode 100644 arch/arm/include/asm/arch-s32/soc.h

diff --git a/arch/arm/cpu/armv8/s32/cpu.c b/arch/arm/cpu/armv8/s32/cpu.c
index 07daca55c0..bffbc35946 100644
--- a/arch/arm/cpu/armv8/s32/cpu.c
+++ b/arch/arm/cpu/armv8/s32/cpu.c
@@ -11,7 +11,6 @@
 #include <asm/system.h>
 #include <asm-generic/sections.h>
 #include "scmi_reset_agent.h"
-#include <asm/arch/soc.h>
 #include <asm-generic/sections.h>
 #include <linux/sizes.h>
 #if defined(CONFIG_DEBUG_UART)
diff --git a/arch/arm/cpu/armv8/s32/s32-gen1/Makefile b/arch/arm/cpu/armv8/s32/s32-gen1/Makefile
index 82c3f47c75..0f821edb9e 100644
--- a/arch/arm/cpu/armv8/s32/s32-gen1/Makefile
+++ b/arch/arm/cpu/armv8/s32/s32-gen1/Makefile
@@ -9,7 +9,6 @@ obj-$(CONFIG_NXP_S32G3XX)		+= s32g3_cmu.o
 obj-$(CONFIG_NXP_S32R45)		+= s32r45_cmu.o
 obj-$(CONFIG_HSE_SECBOOT)		+= hse_adv_secboot.o
 obj-y					+= soc.o
-obj-$(CONFIG_NXP_S32G2XX)		+= s32g274a.o
 ifdef CONFIG_S32_GEN1
 obj-$(CONFIG_NETDEVICES)		+= sgmii/sgmii.o sgmii/serdes_xpcs.o
 endif
diff --git a/arch/arm/cpu/armv8/s32/s32-gen1/s32g274a.c b/arch/arm/cpu/armv8/s32/s32-gen1/s32g274a.c
deleted file mode 100644
index e9b8b819ca..0000000000
--- a/arch/arm/cpu/armv8/s32/s32-gen1/s32g274a.c
+++ /dev/null
@@ -1,60 +0,0 @@
-// SPDX-License-Identifier:     GPL-2.0+
-/*
- * Copyright 2021-2022 NXP
- */
-#include <common.h>
-#include <dm.h>
-#include <linux/sizes.h>
-#include <misc.h>
-#include <s32gen1_siul2_nvram.h>
-
-enum s32g2_derivative {
-	S32G274A_DERIV,
-	S32G254A_DERIV,
-	S32G233A_DERIV,
-	S32G2_INVAL_DERIV,
-};
-
-static inline int get_s32g2_derivative(void)
-{
-	struct udevice *siul20_nvmem = NULL;
-	u32 deriv = 0;
-	int ret;
-
-	ret = uclass_get_device_by_name(UCLASS_MISC, "siul2_0_nvram",
-					&siul20_nvmem);
-	if (ret < 0)
-		return ret;
-
-	ret = misc_read(siul20_nvmem, S32GEN1_SOC_PART_NO, &deriv,
-			sizeof(deriv));
-	if (ret < 0)
-		return ret;
-
-	switch (deriv) {
-	case 0x1D12U:
-		return S32G274A_DERIV;
-	case 0x1CFEU:
-		return S32G254A_DERIV;
-	case 0x1CE9U:
-		return S32G233A_DERIV;
-	};
-
-	pr_err("Invalid S32G2 derivative: 0x%x\n", deriv);
-	return S32G2_INVAL_DERIV;
-}
-
-u32 get_sram_size(void)
-{
-	switch (get_s32g2_derivative()) {
-	case S32G274A_DERIV:
-	case S32G254A_DERIV:
-		return S32_SRAM_SIZE;
-	case S32G233A_DERIV:
-		return 6 * SZ_1M;
-	default:
-		return 0;
-	}
-
-	return 0;
-}
diff --git a/arch/arm/cpu/armv8/s32/s32-gen1/soc.c b/arch/arm/cpu/armv8/s32/s32-gen1/soc.c
index 649ad35b1c..da6c1b11be 100644
--- a/arch/arm/cpu/armv8/s32/s32-gen1/soc.c
+++ b/arch/arm/cpu/armv8/s32/s32-gen1/soc.c
@@ -5,7 +5,6 @@
 
 #include <common.h>
 #include <asm/io.h>
-#include <asm/arch/soc.h>
 #include <mmc.h>
 #include <div64.h>
 #include <errno.h>
diff --git a/arch/arm/include/asm/arch-s32/soc.h b/arch/arm/include/asm/arch-s32/soc.h
deleted file mode 100644
index 1a892c518c..0000000000
--- a/arch/arm/include/asm/arch-s32/soc.h
+++ /dev/null
@@ -1,20 +0,0 @@
-/* SPDX-License-Identifier: GPL-2.0+ */
-/*
- * Copyright 2015 Freescale Semiconductor
- * Copyright 2017-2018,2020-2022 NXP
- */
-#ifndef __ARCH_S32_SOC_H
-#define __ARCH_S32_SOC_H
-
-#include "s32-gen1/s32-gen1-regs.h"
-#include <asm/arch/clock.h>
-#include <asm/arch/mmdc.h>
-
-u32 get_sram_size(void);
-
-static inline bool is_addr_in_sram(uintptr_t addr)
-{
-	return addr >= S32_SRAM_BASE && addr < S32_SRAM_BASE + get_sram_size();
-}
-
-#endif /* __ARCH_S32_SOC_H */
diff --git a/board/freescale/s32-gen1/common.c b/board/freescale/s32-gen1/common.c
index 3bf189cd39..4416f31eb2 100644
--- a/board/freescale/s32-gen1/common.c
+++ b/board/freescale/s32-gen1/common.c
@@ -6,7 +6,6 @@
 #include <common.h>
 #include <asm/io.h>
 #include <asm/arch/clock.h>
-#include <asm/arch/soc.h>
 #include <fdt_support.h>
 #include <linux/libfdt.h>
 #include <miiphy.h>
diff --git a/board/freescale/s32-gen1/eth.c b/board/freescale/s32-gen1/eth.c
index 288edf07a0..535b0e1a32 100644
--- a/board/freescale/s32-gen1/eth.c
+++ b/board/freescale/s32-gen1/eth.c
@@ -12,7 +12,6 @@
 #include <phy.h>
 #include <malloc.h>
 #include <asm/types.h>
-#include <asm/arch/soc.h>
 #if CONFIG_IS_ENABLED(DWC_ETH_QOS_S32CC)
 #include <dm/platform_data/dwc_eth_qos_dm.h>
 #endif
diff --git a/board/freescale/s32-gen1/s32g274abluebox3.c b/board/freescale/s32-gen1/s32g274abluebox3.c
index fb76afd619..e6f248d129 100644
--- a/board/freescale/s32-gen1/s32g274abluebox3.c
+++ b/board/freescale/s32-gen1/s32g274abluebox3.c
@@ -2,7 +2,6 @@
 /*
  * Copyright 2018-2022 NXP
  */
-#include <asm/arch/soc.h>
 #include <board_common.h>
 #include <common.h>
 #include <miiphy.h>
diff --git a/board/freescale/s32-gen1/s32gxxxaevb.c b/board/freescale/s32-gen1/s32gxxxaevb.c
index 1c14b3f567..5cbcccbd48 100644
--- a/board/freescale/s32-gen1/s32gxxxaevb.c
+++ b/board/freescale/s32-gen1/s32gxxxaevb.c
@@ -2,7 +2,6 @@
 /*
  * Copyright 2018-2022 NXP
  */
-#include <asm/arch/soc.h>
 #include <board_common.h>
 #include <common.h>
 #include <dm/uclass.h>
diff --git a/board/freescale/s32-gen1/s32gxxxardb.c b/board/freescale/s32-gen1/s32gxxxardb.c
index f2871ad4ec..985196a748 100644
--- a/board/freescale/s32-gen1/s32gxxxardb.c
+++ b/board/freescale/s32-gen1/s32gxxxardb.c
@@ -2,7 +2,6 @@
 /*
  * Copyright 2018-2022 NXP
  */
-#include <asm/arch/soc.h>
 #include <board_common.h>
 #include <common.h>
 #include <dm/uclass.h>
diff --git a/board/freescale/s32-gen1/s32r45evb.c b/board/freescale/s32-gen1/s32r45evb.c
index 440f987e3d..594ef5cce7 100644
--- a/board/freescale/s32-gen1/s32r45evb.c
+++ b/board/freescale/s32-gen1/s32r45evb.c
@@ -3,7 +3,6 @@
  * Copyright 2018-2022 NXP
  */
 
-#include <asm/arch/soc.h>
 #include <common.h>
 #include <linux/err.h>
 #include <miiphy.h>
diff --git a/drivers/net/dwc_eth_qos_s32cc.c b/drivers/net/dwc_eth_qos_s32cc.c
index a15e8050f3..668e7136f3 100644
--- a/drivers/net/dwc_eth_qos_s32cc.c
+++ b/drivers/net/dwc_eth_qos_s32cc.c
@@ -23,7 +23,6 @@
 #include <reset.h>
 #include <wait_bit.h>
 #include <asm/arch/clock.h>
-#include <asm/arch/soc.h>
 #include <asm/io.h>
 
 #include "board_common.h"
diff --git a/drivers/net/pfeng/pfeng.h b/drivers/net/pfeng/pfeng.h
index 12be3c3c2f..cff0811872 100644
--- a/drivers/net/pfeng/pfeng.h
+++ b/drivers/net/pfeng/pfeng.h
@@ -60,6 +60,9 @@ enum {
 #define PFENG_MODE_DEFAULT PFENG_MODE_ENABLE
 
 /* MC_ME partition definitions */
+#define MC_ME_BASE_ADDR				0x40088000
+#define MC_RGM_BASE_ADDR			0x40078000
+
 #define UPTR(PTR)				((uintptr_t)(PTR))
 #define MC_ME_PRTN_N(MC_ME, n)			(UPTR(MC_ME) + 0x100 + \
 						 (n) * 0x200)
@@ -91,6 +94,7 @@ enum {
 #define PRST_PFE			128
 
 /* Reset domain definitions */
+#define RDC_BASE_ADDR				0x40080000
 #define RDC_RD_N_CTRL(RDC, N)			(UPTR(RDC) + (0x4 * (N)))
 #define RDC_RD_N_STATUS(RDC, N)			(UPTR(RDC) + 0x80 + (0x4 * (N)))
 #define RD_CTRL_UNLOCK_MASK			(0x80000000)
diff --git a/drivers/net/pfeng/pfeng_cmd.c b/drivers/net/pfeng/pfeng_cmd.c
index 200e17d890..1b93605f91 100644
--- a/drivers/net/pfeng/pfeng_cmd.c
+++ b/drivers/net/pfeng/pfeng_cmd.c
@@ -9,7 +9,6 @@
 #include <command.h>
 #include <phy.h>
 #include <asm/arch/clock.h>
-#include <asm/arch/soc.h>
 #include <asm/io.h>
 
 #include <asm/arch/s32-gen1/serdes_regs.h>
-- 
2.17.1

